### PR TITLE
Improve docs for `borg list` --format

### LINKED ISSUE #6061: borg list --format: improve docs, extend placeholders
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

ISSUE

## System information. For client/server mode post info for both machines.

borg version 1.1.17

## Describe the problem you're observing.

The format for `borg-list --format` is barely documented:
There are keys, and from the example I can see that adding `:N` will right-pad the key to length `N`. But what about trimming or left-padding or...? Currently I can't even know what is supported.

Furthermore, the default given applies only to file listings. The default for repo listing should be given as well, which might prevent confusion such as in #3399.

Finally, I would like some extra keys for repo listing:
- size of the archive files (compress/uncompressed) - ignore deduplication for this
- number of files in the archive

### PR DESCRIPTION
**Pull request by hexagonrecursion**
Fixes #6061

I have added 3 examples of formatting usage and a link to the documentation for the full syntax.

See also  #6079

### CODE REVIEW

File: src/borg/archiver.py [hunk 1]
```diff
@@ -4223,16 +4223,41 @@ def define_borg_mount(parser):
...
+        Examples:
+        ::
+
+            $ borg list --format '{archive}{NL}' /path/to/repo
+            ArchiveFoo
+            ArchiveBar
+            ...
+
+            # {VAR:NUMBER} - pad to NUMBER columns.
+            # Note: most fields are left-aligned, but some are right-aligned.
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:4241]: Can this be told more precisely?

Like strings are so, number are so. Or by giving the default format string? Or by pointing at the help output, which display the default format string.

File: src/borg/archiver.py [hunk 2]
```diff
@@ -4223,16 +4223,41 @@ def define_borg_mount(parser):
...
+            $ borg list --format '{path:20} {mode} {uid:20}{NL}' /path/to/repo::ArchiveFoo
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:4242]: Maybe this example could just show the default format string, given explicitly.

File: src/borg/archiver.py [hunk 3]
```diff
@@ -4223,16 +4223,41 @@ def define_borg_mount(parser):
...
 
         """) + BaseFormatter.keys_help() + textwrap.dedent("""
 
-        Keys for listing repository archives:
+        Keys available only when listing archives in a repository:
 
         """) + ArchiveFormatter.keys_help() + textwrap.dedent("""
 
-        Keys for listing archive files:
+        Keys available only when listing files:
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:4260]: ... listing files in an archive:

File: src/borg/archiver.py (hunk 2 above)
Comment (ThomasWaldmann) [reply src/borg/archiver.py:4242]: Or even give 2 examples, one for archive, one for repo listing.

**Comment by hexagonrecursion** [comment]
Idea: I could use pipe instead of a space as a filed separator in examples to illustrate field alignment more clearly:
```sh
$ borg list --format '{mode}|{user:6}|{group:6}|{size:8}|{mtime}|{path}|{extra}{NL}' /path/to/repo::ArchiveFoo 
-rw-rw-r--|user  |user  |    1024|Thu, 2021-12-09 10:22:17|file-foo|
```

**Comment by hexagonrecursion** [comment]
What purpose does `<` serve in the default archive format? The output looks identical without it:
```
$ borg list --format '{archive:<36} {time} [{id}]{NL}' /path/to/repo
ArchiveFoo                           Thu, 2021-12-09 10:22:28 [0b8e9a312bef3f2f6e2d0fc110c196827786c15eba0188738e81697a7fa3b274]
$ borg list --format '{archive:36} {time} [{id}]{NL}' /path/to/repo
ArchiveFoo                           Thu, 2021-12-09 10:22:28 [0b8e9a312bef3f2f6e2d0fc110c196827786c15eba0188738e81697a7fa3b274]
```

**Review by ThomasWaldmann — APPROVED** [review]

**Comment by ThomasWaldmann** [comment]
Thanks! Backport to 1.1-maint?

**Comment by hexagonrecursion** [comment]
> Backport to 1.1-maint?

Sure.

**Pull request merged**